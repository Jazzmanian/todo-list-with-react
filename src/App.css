* {
  margin: 0;
  padding: 0;
  font-family: sans-serif;
}
.complete {
  text-decoration: line-through;
}

.App {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  background-color: #141a2f;
}
.title {
  color: white;
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.form {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-bottom: auto;
  margin-bottom: 0rem;
  margin-top: 0rem;
}

input {
  border: 0;
  outline: 0;
  padding: 5px;
  border: solid 4px rgb(248, 91, 0);
  background-color: #141a2f;
  color: white;
  border-top-left-radius: 25px;
  border-bottom-left-radius: 25px;
}
.form button {
  border-top-right-radius: 25px;
  border-bottom-right-radius: 25px;
  background-color: rgb(248, 91, 0);
  border: 0;
  outline: 0;
  color: white;
  padding: 9px;
}
.elements {
  display: flex;
  align-items: center;
  flex-direction: column;
  margin-top: 2rem;
}
.elements div {
  margin-left: 2rem;
  margin-right: 2rem;
  margin-top: 1rem;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0.7rem;
  border-radius: 0.3rem;
  background: hsla(186, 100%, 69%, 1);
  color: rgb(255, 255, 255);
  background: hsla(217, 100%, 50%, 1);

  background: linear-gradient(
    0deg,
    hsla(217, 100%, 50%, 1) 0%,
    hsla(186, 100%, 69%, 1) 100%
  );

  background: -moz-linear-gradient(
    0deg,
    hsla(217, 100%, 50%, 1) 0%,
    hsla(186, 100%, 69%, 1) 100%
  );

  background: -webkit-linear-gradient(
    0deg,
    hsla(217, 100%, 50%, 1) 0%,
    hsla(186, 100%, 69%, 1) 100%
  );

  filter: progid: DXImageTransform.Microsoft.gradient( startColorstr="#0061FF", endColorstr="#60EFFF", GradientType=1 );
}
