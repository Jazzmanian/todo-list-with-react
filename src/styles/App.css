* {
  margin: 0;
  padding: 0;
  font-family: sans-serif;
}
.complete {
  text-decoration: line-through;
}

.App {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  background-color: #141a2f;
}
.title {
  color: white;
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.form {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-bottom: auto;
  margin-bottom: 0rem;
  margin-top: 0rem;
}

input {
  border: 0;
  outline: 0;
  padding: 5px;
  border: solid 4px rgb(248, 91, 0);
  background-color: #141a2f;
  color: white;
  border-top-left-radius: 25px;
  border-bottom-left-radius: 25px;
  margin-left: 60px;
}
.form button {
  border-top-right-radius: 25px;
  border-bottom-right-radius: 25px;
  background-color: rgb(248, 91, 0);
  border: 0;
  outline: 0;
  color: white;
  padding: 9px;
}
.elements {
  display: flex;
  align-items: center;
  flex-direction: column;
  margin-top: 2rem;
}
.todo-items {
  margin-left: 2rem;
  margin-right: 2rem;
  margin-top: 1rem;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0.7rem;
  border-radius: 0.3rem;
  background: hsla(186, 100%, 69%, 1);
  color: rgb(255, 255, 255);
  background: hsla(217, 100%, 50%, 1);

  background: linear-gradient(
    0deg,
    hsla(217, 100%, 50%, 1) 0%,
    hsla(186, 100%, 69%, 1) 100%
  );

  background: -moz-linear-gradient(
    0deg,
    hsla(217, 100%, 50%, 1) 0%,
    hsla(186, 100%, 69%, 1) 100%
  );

  background: -webkit-linear-gradient(
    0deg,
    hsla(217, 100%, 50%, 1) 0%,
    hsla(186, 100%, 69%, 1) 100%
  );

  filter: progid: DXImageTransform.Microsoft.gradient( startColorstr="#0061FF", endColorstr="#60EFFF", GradientType=1 );
}
.todo-items h4 {
  padding-right: 10rem;
}
.todo-items button {
  font-size: 120%;
  background-color: inherit;
  border: 0;
}
