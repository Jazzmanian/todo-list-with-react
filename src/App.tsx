import React, { useState } from 'react';
import './styles/App.css';
import TodoForm from './components/TodoForm';
import TodoList from './components/TodoList';
import { ITask, AddTodo } from './types';

const App: React.FC = () => {
  const [taskList, setTaskList] = useState<ITask[]>([]);

  const addTodo: AddTodo = (newTodo: string) => {
    let id = 1;
    if (taskList.length > 0) {
      id = taskList.length + 1;
    }
    const newTask = {
      id,
      name: newTodo,
      completed: false,
    };

    setTaskList([newTask, ...taskList]);
  };

  return (
    <div className="App">
      <h1 className="title">Todo List</h1>
      <div className="form">
        <TodoForm addTodo={addTodo} />
        <TodoList taskList={taskList} />
      </div>
    </div>
  );
};

export default App;
