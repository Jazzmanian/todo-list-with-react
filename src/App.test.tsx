import { render, screen } from '@testing-library/react';
import App from './App';
import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

describe('App test', () => {
  it('should render app', () => {
    render(<App />);
    expect(screen.getByText(/Todo list/i)).toBeInTheDocument();
  });
  it('should render the snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
