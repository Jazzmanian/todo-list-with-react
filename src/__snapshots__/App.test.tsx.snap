// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test should render the snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="title"
  >
    Todo List
  </h1>
  <div
    className="form"
  >
    <div>
      <div>
        <input
          aria-label="todo-input"
          autoFocus={true}
          onChange={[Function]}
          placeholder="Enter your todo item"
          type="text"
          value=""
        />
        <button
          onClick={[Function]}
        >
          Add Items
        </button>
        <label
          className="todo-tasks-container"
        />
      </div>
    </div>
  </div>
</div>
`;
